modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/AddressCommerceCheckoutStepUtil.java

```diff
@@ -201,6 +212,22 @@ private CommerceAddress _addCommerceAddress(
 			CommerceOrder commerceOrder, ActionRequest actionRequest)
 		throws Exception {
 
+		long countryId = ParamUtil.getLong(actionRequest, "countryId");
+
+		boolean useAsBilling = ParamUtil.getBoolean(
+			actionRequest, "use-as-billing");
+
+		if (useAsBilling) {
+			_commerceAddressType =
+				CommerceAddressConstants.ADDRESS_TYPE_BILLING_AND_SHIPPING;
+
+			Country country = _countryLocalService.getCountry(countryId);
+
+			if (!country.isBillingAllowed()) {
```

Comment:
this should be done before assigning a value to _commerceAddressType, so that we throw the exception before any other instruction.

Reply:
See commit: https://github.com/liferay-commerce/liferay-portal/pull/3448/commits/75874b1a13a8f646199910022c9e0e3fd58fad24